### PR TITLE
feat(cli): file system prettier migration 

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

When we migrate the configuration from Prettier, I decided to **not** run the other migrations. What do you think? Should we run them at the end?

This PR adds the required operations for the migration from prettier:
- file system operations
- workspace operations
- merging strategy to the main configuration

## Test Plan

I added new tests cases, errors too

### CODE REVIEW

File: crates/biome_cli/src/execute/migrate.rs [hunk 1]
```diff
@@ -34,9 +42,13 @@ pub(crate) fn run(migrate_payload: MigratePayload) -> Result<(), CliDiagnostic>
         configuration_directory_path,
         verbose,
         // we will use it later
-        prettier: _,
+        prettier,
```
Comment (arendjr) [inline crates/biome_cli/src/execute/migrate.rs:45]: Looks like the comment can go.

File: crates/biome_cli/tests/snapshots/main_commands_migrate/prettier_migrate.snap [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  i Configuration file can be updated.
+  
+    1    │ - {·"linter":·{·"enabled":·true·}·}
+       1 │ + {
+       2 │ + → "formatter":·{
+       3 │ + → → "enabled":·true,
+       4 │ + → → "formatWithErrors":·false,
+       5 │ + → → "indentStyle":·"space",
+       6 │ + → → "indentWidth":·4,
```
Comment (arendjr) [inline crates/biome_cli/tests/snapshots/main_commands_migrate/prettier_migrate.snap:30]: I believe Prettier’s default indent width is 2: https://prettier.io/docs/en/options.html#tab-width

**Comment by arendjr** [comment]
> When we migrate the configuration from Prettier, I decided to not run the other migrations. What do you think? Should we run them at the end?

Just for my understanding, what other migrations are those? Intuitively I’d say it makes sense to only run one at a time, but I’m lacking of context :)

**Comment by ematipico** [comment]
> > When we migrate the configuration from Prettier, I decided to not run the other migrations. What do you think? Should we run them at the end?
> 
> Just for my understanding, what other migrations are those? Intuitively I’d say it makes sense to only run one at a time, but I’m lacking of context :)

At the moment, we only have a migration that updates the version of the schema to the latest.

**Comment by Conaclos** [comment]
> When we migrate the configuration from Prettier, I decided to not run the other migrations. What do you think? Should we run them at the end?

I think it is better to not to?

By the way, why is this feature is not under the `init` command? `migrate` is dedicated to a migration between two biome versions?

**Comment by arendjr** [comment]
> At the moment, we only have a migration that updates the version of the schema to the latest.

Yeah, agreed with your approach 👍

**Comment by arendjr** [comment]
> By the way, why is this feature is not under the `init` command? `migrate` is dedicated to a migration between two biome versions?

That’s a good point. I can kinda see the use case for having it under `migrate`, but it would be really cool if `init` would check for the presence of Prettier config too and would prompt interactively if you want to initiate the migration.

**Comment by ematipico** [comment]
> By the way, why is this feature is not under the init command? migrate is dedicated to a migration between two biome versions?

Because a user would want to migrate FROM prettier, not init the project with Prettier's defaults. 

That was how my brain thinks 😁

**Comment by ematipico** [comment]
> Or if interactive prompts are a bit too far-fetched, just a notice diagnostic to inform users of the presence of the migrate command for this.

I don't think they are far-fetched. I am looking forward to use these prompts: https://crates.io/crates/cliclack 🤩

**Comment by ematipico** [comment]
I am going to merge this. Feel free to add more feedback if you need it. The feature isn't finished though, so I will address possible concerns in the next PR

**Comment by Conaclos** [comment]
> > Or if interactive prompts are a bit too far-fetched, just a notice diagnostic to inform users of the presence of the migrate command for this.
> 
> I don't think they are far-fetched. I am looking forward to use these prompts: https://crates.io/crates/cliclack 🤩

Giving my conservative cent: if it is only used in a few places, we should avoid adding an extra dependency / way of interacting with Biome.

**Pull request merged**